app/components/primer/layout.rb

```diff
@@ -0,0 +1,347 @@
+# frozen_string_literal: true
+
+module Primer
+  # Add a general description of component here
+  # Add additional usage considerations or best practices that may aid the user to use the component correctly.
+  # @accessibility Add any accessibility considerations
+  class Layout < Primer::Component
+    CONTAINER_DEFAULT = :full
+    CONTAINER_OPTIONS = [:full, :xl, :lg, :md].freeze
+
+    DENSITY_DEFAULT = :none
+    DENSITY_MAPPINGS = {
+      DENSITY_DEFAULT => 0,
+      :compact => 3,
+      :normal => [nil, 3, nil, 4],
+      :relaxed => [nil, 3, nil, 4, 5]
+    }.freeze
+    DENSITY_OPTIONS = DENSITY_MAPPINGS.keys.freeze
+
+    SIDEBAR_WIDTH_DEFAULT = :default
+    SIDEBAR_WIDTH_MAPPINGS = {
+      SIDEBAR_WIDTH_DEFAULT => "",
+      :narrow => "Layout--sidebar-narrow",
+      :wide => "Layout--sidebar-wide"
+    }.freeze
+    SIDEBAR_WIDTH_OPTIONS = SIDEBAR_WIDTH_MAPPINGS.keys.freeze
+
+    GUTTER_DEFAULT = :default
+    GUTTER_MAPPINGS = {
+      GUTTER_DEFAULT => "",
+      :none => "Layout--gutter-none",
+      :condensed => "Layout--gutter-condensed",
+      :spacious => "Layout--gutter-spacious"
+    }.freeze
+    GUTTER_OPTIONS = GUTTER_MAPPINGS.keys.freeze
+
+    SIDEBAR_PLACEMENT_DEFAULT = :start
+    SIDEBAR_PLACEMENT_OPTIONS = [SIDEBAR_PLACEMENT_DEFAULT, :end].freeze
+
+    SIDEBAR_FLOW_ROW_PLACEMENT_DEFAULT = :start
+    SIDEBAR_FLOW_ROW_PLACEMENT_OPTIONS = [SIDEBAR_FLOW_ROW_PLACEMENT_DEFAULT, :end, :none].freeze
+
+    FLOW_ROW_UNTIL_DEFAULT = :sm
+    FLOW_ROW_UNTIL_MAPPINGS = {
+      FLOW_ROW_UNTIL_DEFAULT => "",
+      :md => "Layout--flowRow-until-md",
+      :lg => "Layout--flowRow-until-lg"
+    }.freeze
+    FLOW_ROW_UNTIL_OPTIONS = FLOW_ROW_UNTIL_MAPPINGS.keys.freeze
+
+    MAIN_WIDTH_DEFAULT = :full
+    MAIN_WIDTH_OPTIONS = [MAIN_WIDTH_DEFAULT, :md, :lg, :xl].freeze
+
+    DIVIDER_FLOW_ROW_VARIANT_DEFAULT = :visible
+    DIVIDER_FLOW_ROW_VARIANT_MAPPINGS = {
+      DIVIDER_FLOW_ROW_VARIANT_DEFAULT => "",
+      :hidden => "Layout-divider--flowRow-hidden",
+      :shallow => "Layout-divider--flowRow-shallow"
+    }.freeze
+    DIVIDER_FLOW_ROW_VARIANT_OPTIONS = DIVIDER_FLOW_ROW_VARIANT_MAPPINGS.keys
+
+    # The layout's main content.
+    #
+    # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+    renders_one :main, ->(**system_arguments) { Primer::Layout::Main.new(width: @main_width, **system_arguments) }
+
+    # The layout's sidebar.
+    #
+    # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+    renders_one :sidebar, lambda { |**system_arguments|
+      system_arguments[:tag] = :div
+      system_arguments[:classes] = class_names(
+        "Layout-sidebar",
+        system_arguments[:classes]
+      )
+
+      Primer::BaseComponent.new(**system_arguments)
+    }
+
+    # @example Default
+    #
+    #   <%= render(Primer::Layout.new) do |c| %>
+    #     <% c.main(border: true) { "Main" } %>
```

Comment:
What do you think about having an example with _only_ defaults? (so no border)

Reply:
(Or do we want `border` to default to `true`?)

Reply:
I'm adding borders to the examples so it is easier to **see** the sidebar and main blocks. Otherwise it'd just be text in the page.

- with border

![image](https://user-images.githubusercontent.com/11280312/120512786-43cb4000-c391-11eb-9070-8ceaf963f9cc.png)

- without border

![image](https://user-images.githubusercontent.com/11280312/120512839-4f1e6b80-c391-11eb-8746-da5ea2d2ea4c.png)